homeassistant/components/ruuvi_gateway/config_flow.py

```diff
@@ -0,0 +1,82 @@
+"""Config flow for Ruuvi Gateway integration."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+import aioruuvigateway.api as gw_api
+from aioruuvigateway.excs import CannotConnect, InvalidAuth
+
+from homeassistant import config_entries
+from homeassistant.components import dhcp
+from homeassistant.const import CONF_HOST, CONF_TOKEN
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers.httpx_client import get_async_client
+
+from . import DOMAIN
+from .schemata import CONFIG_SCHEMA, get_config_schema_with_default_host
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Ruuvi Gateway."""
+
+    VERSION = 1
+
+    def __init__(self) -> None:
+        """Initialize the config flow."""
+        super().__init__()
+        self.config_schema = CONFIG_SCHEMA
+
+    async def _async_validate(
+        self,
+        user_input: dict[str, Any],
+    ) -> tuple[FlowResult | None, dict[str, str]]:
+        """Validate configuration (either discovered or user input)."""
+        errors: dict[str, str] = {}
+
+        try:
+            async with get_async_client(self.hass) as client:
+                resp = await gw_api.get_gateway_history_data(
+                    client,
+                    host=user_input[CONF_HOST],
+                    bearer_token=user_input[CONF_TOKEN],
+                )
+            await self.async_set_unique_id(resp.gw_mac, raise_on_progress=False)
+            self._abort_if_unique_id_configured()
```

Comment:
We should update the host so they don't have to reconfigure if the up changes

Reply:
I'm sorry, I'm not quite following?

Reply:
`self._abort_if_unique_id_configured(updates={CONF_ADDRESS: host})`